core/src/main/scala/akka/persistence/jdbc/tag/EventTagConverter.scala

```diff
@@ -0,0 +1,59 @@
+package akka.persistence.jdbc.tag
+
+import java.util.concurrent.ConcurrentHashMap
+
+import akka.persistence.jdbc.db.PostgresErrorCodes
+import org.postgresql.util.PSQLException
+import slick.jdbc.JdbcBackend._
+
+import scala.annotation.tailrec
+import scala.concurrent.{ExecutionContext, Future}
+
+trait EventTagConverter {
+  def getIdByName(name: String): Future[Int]
+  def getIdByNameForce(name: String): Int
+}
+
+class EventTagDao(db: Database)(implicit ctx: ExecutionContext) extends EventTagConverter {
+  val queries = new EventTagQueries()
+
+  // TODO should we load on startup??
+  // the biggest problem with cache like below is that we only store new values, so you can have memory leak, shouldn't we delete some old event?
+  // maybe we should cache in different way?
```

Comment:
Ohh.. my todo :) so what do you think? Load on startup or not?, Is it ok form of caching?

Reply:
I'm going to address this in a separate PR. Also I thought about using some data structure that supports TTL for the cache field.